### PR TITLE
Animation Fixes.

### PR DESCRIPTION
**Pull request by ludvigarlebrink**

### CODE REVIEW

File: shaders/Skinning.vert [hunk 1]
```diff
@@ -35,9 +35,9 @@ void main () {
     normal += (bones[vertexBoneIDs[2]] * vec4(vertexNormal, 0.0)) * vertexWeights[2];
     normal += (bones[vertexBoneIDs[3]] * vec4(vertexNormal, 0.0)) * vertexWeights[3];
     
-    gl_Position = viewProjection * (model * position);
-    vertexOut.pos = (model * position).xyz;
-    vertexOut.normal = normalize(normalMatrix * normal.xyz);
```
Comment (andens) [inline shaders/Skinning.vert]: The `normalMatrix` uniform is no longer used, so the variable could be removed.

File: src/Video/RenderProgram/SkinRenderProgram.cpp [hunk 2]
```diff
@@ -196,8 +196,9 @@ void SkinRenderProgram::Render(const Geometry::Geometry3D* geometry, const Textu
         // Render model.
         glUniformMatrix4fv(modelLocation, 1, GL_FALSE, &modelMatrix[0][0]);
         glm::mat4 normalMatrix = glm::transpose(glm::inverse(viewMatrix * modelMatrix));
```
Comment (andens) [inline src/Video/RenderProgram/SkinRenderProgram.cpp]: This one is no longer used in the shader.

File: src/Engine/Animation/AnimationController.cpp [hunk 3]
```diff
@@ -32,21 +32,29 @@ void AnimationController::Save(const std::string& path) {
         if (typeid(*node) == typeid(AnimationAction)) {
             NodeType nodetype = ACTION;
             file.write(reinterpret_cast<char*>(&nodetype), sizeof(NodeType));
+            AnimationAction * animationAction = dynamic_cast<AnimationAction *>(node);
```
Comment (andens) [inline src/Engine/Animation/AnimationController.cpp]: Both asterisks should be left-aligned.

File: src/Engine/Animation/AnimationController.cpp [hunk 4]
```diff
@@ -32,21 +32,29 @@ void AnimationController::Save(const std::string& path) {
...
+
+            if (animationAction)
+                animationAction->Save(&file);
         } else {
             NodeType nodetype = TRANSITION;
             file.write(reinterpret_cast<char*>(&nodetype), sizeof(NodeType));
-        }
+            AnimationTransition * animationTransition = dynamic_cast<AnimationTransition *>(node);
```
Comment (andens) [inline src/Engine/Animation/AnimationController.cpp]: Both asterisks should be left-aligned.

File: src/Engine/Util/Node.hpp [hunk 5]
```diff
@@ -32,5 +37,10 @@ struct Node {
         file->read(reinterpret_cast<char*>(&index), sizeof(uint32_t));
         file->read(reinterpret_cast<char*>(name), 128);
         file->read(reinterpret_cast<char*>(&numInputSlots), sizeof(uint32_t));
+        file->read(reinterpret_cast<char *>(inputIndex), sizeof(uint32_t) * numInputSlots);
```
Comment (andens) [inline src/Engine/Util/Node.hpp]: Misaligned asterisk.

File: src/Engine/Util/Node.hpp [hunk 6]
```diff
@@ -32,5 +37,10 @@ struct Node {
...
+        file->read(reinterpret_cast<char*>(&numOutputSlots), sizeof(uint32_t));
+        file->read(reinterpret_cast<char *>(outputIndex), sizeof(uint32_t) * numOutputSlots);
```
Comment (andens) [inline src/Engine/Util/Node.hpp]: Misaligned asterisk.

File: shaders/Skinning.vert (hunk 1 above)
Comment (ludvigarlebrink) [reply shaders/Skinning.vert]: @Ittis  is looking into this.

File: src/Video/RenderProgram/SkinRenderProgram.cpp (hunk 2 above)
Comment (ludvigarlebrink) [reply src/Video/RenderProgram/SkinRenderProgram.cpp]: @Ittis  is looking into this.

**Pull request merged**